src/MessagingAPI.ts

```diff
@@ -39,6 +39,9 @@ export interface MessagingAPI {
     /** Listen to new messages */
     onMessage(listener: (conversation: Conversation, message: TextMessage) => void): void
 
+    /** Listen to updates on the membership of a channel */
+    onChannelMembership(listener: (conversation: Conversation, membership: string) => void): void
```

Comment:
Do we know the actual `membership` events universe? If so I would use a type here (sth like `type Membership = 'join' | 'another' | ...`), if that's not the case, just disregard this message.

Reply:
I'm leaving it as a string and adding documentation.